src/Tempest/Http/src/Routing/Construction/RouteConfigConstructor.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Http\Routing\Construction;
+
+use Tempest\Container\Singleton;
+use Tempest\Http\Route;
+use Tempest\Http\RouteConfig;
+
+/**
+ * @internal
+ */
+#[Singleton]
+final class RouteConfigConstructor
```

Comment:
All logic for building up the routing tree during discovery is moved here, it was originally in the route config. This singleton service will keep the state during construction. And it will result in a `RouteConfig` as output. This can be applied to the DI-ed RouteConfig